### PR TITLE
🦀🧪 Expand driver/unit tests to orbital shaker & vacuum (and snuck in ruff)

### PR DESCRIPTION
**Pull request by JavanW-W**
Initial driver framework in place; all baseline tests pass, more to come

Also added ruff (forgot to make its own branch/PR, sorry)

### CODE REVIEW

File: ika/tests/test_shaker_driver.py [hunk 1]
```diff
@@ -71,23 +63,37 @@ async def test_setpoint_roundtrip():
     """Confirm that the setpoint can be updated."""
     async def get():
         async with Shaker(ADDRESS) as device:
-            with pytest.raises(ValueError):
```
Comment (alexrudd2) [inline ika/tests/test_shaker_driver.py:74]: Don't remove this test!

https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

File: ika/tests/test_shaker_driver.py [hunk 2]
```diff
@@ -71,23 +63,37 @@ async def test_setpoint_roundtrip():
...
 async def test_start_stop():
     """Confirm that the shaker motor can be controlled."""
     async def get():
         async with Shaker(ADDRESS) as device:
-            assert await device.get_running() is False
-            await device.control(on=True)
-            assert await device.get_running() is True
-            await device.control(on=False)
-            assert await device.get_running() is False
+            response = await device.get()
```
Comment (alexrudd2) [inline ika/tests/test_shaker_driver.py:80]: The point of having separate tests here is to test each function.  There is already a test for `get()`, so this test should be confined to `get_running()`.  (This isn't always possible, but is in this case)

File: ika/driver.py [hunk 3]
```diff
@@ -282,23 +282,123 @@ async def reset(self):
...
+        - 1 start bit; 7 character bits; 1 parity bit (even); 1 stop bit
+        - Transmission speed: 9600 bit/s
+
     Command syntax and format from the manual:
         - commands and parameters are transmitted as capital letters
         - commands and parameters including successive parameters are separated by at least
           one space (hex 0x20)
         - each individual command (including parameters and data and each response are terminated
-          with Blank CR LF (hex 0x20 hex 0x0d hex 0x0A) and have a maximum length of 80 characters
+          with Blank CR LF (hex 0x0d hex 0x0A) and have a maximum length of 80 characters
```
Comment (alexrudd2) [inline ika/driver.py:296]: Good catch; the manual is incorrect.  (different manuals say different things, which is annoying).

However, your update isn't fully correct, either.

Blank CR LF != 0x0d 0x0A

File: setup.py [hunk 4]
```diff
@@ -26,9 +26,9 @@
             'pytest-cov>=4,<5',
             'pytest-asyncio==0.*',
             'pytest-xdist==3.*',
-            'flake8==6.*',
-            'flake8-docstrings==1.*',
```
Comment (alexrudd2) [inline setup.py:30]: You'll need to update Azure Pipelines, too

https://github.com/numat/ika/blob/13bf457f4776fe9ed2ea2cd7ea865541541c88b4/azure-pipelines.yml#L33-L35

**Review by alexrudd2 — CHANGES_REQUESTED** [review]

File: ika/tests/test_shaker_driver.py (hunk 2 above)
Comment (JavanW-W) [reply ika/tests/test_shaker_driver.py:80]: Screen Shot 2023-04-10 at 2 42 56 PM
There is no `get_running()`, so I'm confused. The function being tested here is `control()` not `get()`. This is consistent with the drivers you've written.
https://github.com/numat/ika/blob/0b9036303ea75786b3ef86c220c1a4a4f7f30924/ika/tests/test_hotplate_driver.py#L76
https://github.com/numat/ika/blob/0b9036303ea75786b3ef86c220c1a4a4f7f30924/ika/tests/test_stir_driver.py#L84

File: ika/driver.py (hunk 3 above)
Comment (JavanW-W) [reply ika/driver.py:296]: So which is correct, left or right? Do I just need to trial and error to figure it out? (The Vacstar manual says "CRLF (code: hex 0x20 hex 0x0d hex 0x0A)" which is perhaps the most confusing of all)

File: ika/tests/test_shaker_driver.py (hunk 2 above)
Comment (alexrudd2) [reply ika/tests/test_shaker_driver.py:80]: Oh, fair enough.  I got smarter on the `ismatec` driver but never went back and updated `ika`
https://github.com/numat/ismatec/blob/fd98ee74cbf9de71af28a5c21c4c80bed1d5f865/ismatec/tests/test_driver.py#L70

File: ika/driver.py (hunk 3 above)
Comment (alexrudd2) [reply ika/driver.py:296]: Yes, experiment and find out.

**Review by alexrudd2 — APPROVED** [review]

**Pull request merged**